### PR TITLE
[GSOC][TMVA][SOFIE] Extend Concat ONNX Operator to implement stack functionality as the ONNX  ConcatFromSequence operator 

### PR DESCRIPTION
**Pull request by Neel-Shah-29**
# This Pull request: Adds ConcatFromSequence ONNX Operator

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

File: tmva/sofie/CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,7 @@ ROOT_STANDARD_LIBRARY_PACKAGE(ROOTTMVASofie
    TMVA/ROperator_Slice.hxx
    TMVA/ROperator_Transpose.hxx
    TMVA/ROperator_Pool.hxx
```
Comment (lmoneta) [inline tmva/sofie/CMakeLists.txt:32]: I would keep as name `Concat` and not `ConcatFromSequence`

File: tmva/sofie/inc/TMVA/ROperator_ConcatFromSequence.hxx [hunk 2]
```diff
@@ -47,33 +48,40 @@
...
-                 throw std::runtime_error("TMVA SOFIE Concat Op - input tensors have different shapes " +
-                  ConvertShapeToString(inputs[i]) + " and " + ConvertShapeToString(inputs[i-1]));
-               for (size_t iaxis = 0; iaxis < inputs[i].size(); iaxis++) {
-                 if ((int) iaxis == fAxis)
-                  concat_dim += inputs[i][iaxis];
-                 else
-                    if (i> 0 && inputs[i][iaxis] != inputs[i-1][iaxis])
-                     throw std::runtime_error("TMVA SOFIE Concat Op - input tensors have wrong shapes " +
-                      ConvertShapeToString(inputs[i]) + " and " + ConvertShapeToString(inputs[i-1]));
+                  if (i > 0 && inputs[i].size() != inputs[i-1].size() )
```
Comment (lmoneta) [inline tmva/sofie/inc/TMVA/ROperator_ConcatFromSequence.hxx:55]: The indentation here is not correct, keep it as it was before

File: tmva/sofie_parsers/inc/TMVA/RModelParser_ONNX.hxx [hunk 3]
```diff
@@ -85,7 +85,7 @@ const factoryMethodMap mapOptypeOperator = {
    {"Flatten", &make_ROperator_Reshape},
    {"Identity", &make_ROperator_Identity},
    {"Softmax", &make_ROperator_Softmax},
-   {"Concat", &make_ROperator_Concat},
```
Comment (lmoneta) [inline tmva/sofie_parsers/inc/TMVA/RModelParser_ONNX.hxx:88]: Also here keep Concat

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default with flags -Dtmva-sofie=On

**Review by sanjibansg — APPROVED** [review]
LGTM!

**Review by lmoneta — APPROVED** [review]
LGTM!
Thank you @Neel-Shah-29 for this new improvement in Concat. 
I will change maybe just the PR title

**Pull request merged**